### PR TITLE
Performance regression tests for instant execution on Java builds

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

File: subprojects/performance/templates.gradle [hunk 1]
```diff
@@ -79,6 +79,15 @@ tasks.register("generateLotsOfDeprecationWarnings", Copy) {
     from "src/templates/$name"
 }
 
+tasks.register("largeJavaBuildBuilder", BuildBuilderGenerator) {
```
Comment (bamboo) [inline subprojects/performance/templates.gradle]: 🔴 `largeJavaBuildBuilder` seems unrelated.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/JavaTestProject.groovy [hunk 2]
```diff
@@ -30,18 +30,20 @@ enum JavaTestProject {
...
     MEDIUM_JAVA_MULTI_PROJECT_WITH_TEST_NG("mediumJavaMultiProjectWithTestNG", GROOVY, 100, 100, '512m', '256m', true, [assemble: productionFile('mediumJavaMultiProjectWithTestNG'), test: productionFile('mediumJavaMultiProjectWithTestNG', 50, 250, 5000)]),
 
     SMALL_JAVA_MULTI_PROJECT("smallJavaMultiProject", GROOVY, 50, 10, '256m', '64m', false, [assemble: productionFile('smallJavaMultiProject')]),
+    SMALL_JAVA_MULTI_PROJECT_NO_BUILD_SRC("smallJavaMultiProjectNoBuildSrc", GROOVY, false, 50, 10, '256m', '64m', false, [assemble: productionFile('smallJavaMultiProject')]),
 
     private TestProjectGeneratorConfiguration config
 
-    JavaTestProject(String projectName, GradleDsl dsl, CompositeConfiguration compositeConfiguration = null, int sourceFiles, int subProjects, String daemonMemory, String compilerMemory, boolean useTestNG, Map<String, String> filesToUpdate) {
+    JavaTestProject(String projectName, GradleDsl dsl, boolean buildSrc = true, CompositeConfiguration compositeConfiguration = null, int sourceFiles, int subProjects, String daemonMemory, String compilerMemory, boolean useTestNG, Map<String, String> filesToUpdate) {
```
Comment (bamboo) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/JavaTestProject.groovy]: ⭕️ It is a bit unsettling that this works at all :)

File: subprojects/performance/templates.gradle (hunk 1 above)
Comment (eskatos) [reply subprojects/performance/templates.gradle]: Yep, was a leftover, good catch thanks, removed

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/JavaTestProject.groovy [hunk 3]
```diff
@@ -30,18 +30,20 @@ enum JavaTestProject {
...
+    LARGE_JAVA_MULTI_PROJECT_NO_BUILD_SRC("largeJavaMultiProjectNoBuildSrc", GROOVY, false, 100, 500, '1536m', '256m', false, [assemble: productionFile('largeJavaMultiProject'), test: productionFile('largeJavaMultiProject', 450, 2250, 45000)]),
 
     MEDIUM_MONOLITHIC_JAVA_PROJECT("mediumMonolithicJavaProject", GROOVY, 10000, 0, '512m', '1g', false, [assemble: productionFile('mediumMonolithicJavaProject', -1)]),
     MEDIUM_JAVA_MULTI_PROJECT("mediumJavaMultiProject", GROOVY, 100, 100, '512m', '256m', false, [assemble: productionFile('mediumJavaMultiProject')]),
-    MEDIUM_JAVA_COMPOSITE_BUILD("mediumJavaCompositeBuild", GROOVY, composite(false), 100, 100, '768m', '256m', false, [assemble: productionFile('mediumJavaMultiProject')]),
-    MEDIUM_JAVA_PREDEFINED_COMPOSITE_BUILD("mediumJavaPredefinedCompositeBuild", GROOVY, composite(true), 100, 100, '768m',  '256m', false, [assemble: productionFile('mediumJavaMultiProject')]),
+    MEDIUM_JAVA_COMPOSITE_BUILD("mediumJavaCompositeBuild", GROOVY, true, composite(false), 100, 100, '768m', '256m', false, [assemble: productionFile('mediumJavaMultiProject')]),
```
Comment (bamboo) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/JavaTestProject.groovy]: :x: Since `true` is already the default I'd remove it from these two calls.

Comment (eskatos) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/JavaTestProject.groovy]: It can't be removed, Groovy gets confused.
I'll rework the constructor arguments, it's a mess

Comment (eskatos) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/JavaTestProject.groovy]: Done, please take a look
